File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.java
class Solution {
    public int countNegatives(int[][] grid) {
        int n = grid.length;
        int m = grid[0].length;
        int cnt = 0;
        int col = 0;
        int row = n-1;
        
        while( row>=0 && col<m){
            if(grid[row][col]<0){
                row--;
                cnt += m-col;
            }else{
                col++;
            }
        }
        return cnt;
    }
}

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.ts
function countNegatives(grid: number[][]): number {
  const arr: number[] = grid.flat();

  const count: number = arr.reduce((curr: number, next: number) => {
    if (next < 0) {
      return curr + 1;
    } else {
      return curr;
    }
  }, 0);
  return count;
}

